jax/experimental/export/export.py

```diff
@@ -307,52 +308,40 @@ def default_lowering_platform() -> str:
   # Canonicalize to turn 'gpu' into 'cuda' or 'rocm'
   return xb.canonicalize_platform(jax.default_backend())
 
-def poly_spec(
-    arg_shape: Sequence[Optional[int]],
-    arg_dtype: DType,
-    polymorphic_shape: Optional[str]) -> jax.ShapeDtypeStruct:
+def symbolic_shape(
+  shape_spec: Optional[str],
+  *,
+  like: Optional[Sequence[Optional[int]]] = None) -> Shape:
   """Constructs a jax.ShapeDtypeStruct with polymorphic shapes.
 
   Args:
-    arg_shape: the shape, with possibly some unspecified dimensions.
-    arg_dtype: the jax dtype.
-    polymorphic_shape: a string specifying the polymorphic shape.
-
-      .. warning:: The shape-polymorphic lowering is an experimental feature.
-        It is meant to be sound, but it is known to reject some JAX programs
-        that are shape polymorphic. The details of this feature can change.
-
-      It should be either `None` (all dimensions are constant), or a string of
```

Comment:
Do we use [myst](https://myst-parser.readthedocs.io/en/latest/intro.html) for parsing/rendering docstrings?

If not, the default markup language in Sphinx is reStructuredText, which has a slightly different link syntax: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#hyperlinks.